src/Mvc/Mvc.RazorPages/src/Infrastructure/DynamicPageEndpointSelectorCache.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    internal class DynamicPageEndpointSelectorCache
+    {
+        private ConcurrentDictionary<int, EndpointDataSource> _dataSourceCache = new();
+        private ConcurrentDictionary<int, DynamicPageEndpointSelector> _endpointSelectorCache = new();
+
+        public void Dispose()
+        {
+            _dataSourceCache.Clear();
```

Comment:
Why is it important to clear these when the object is being disposed?

Reply:
done